Purchases/Purchasing/TrialOrIntroPriceEligibilityChecker.swift

```diff
@@ -66,6 +67,13 @@ class TrialOrIntroPriceEligibilityChecker {
         }
     }
 
+    func checkEligibility(product: StoreProduct,
+                          completion: @escaping ReceiveIntroEligibilityStatusBlock) {
+        checkEligibility(productIdentifiers: [product.productIdentifier]) { eligibility in
+            completion(eligibility[product.productIdentifier]?.status ?? .unknown)
+        }
+    }
+
```

Comment:
Even though this method's implementation is very straightforward, we should add tests (I imagine that they haven't been added because this is still WIP). 
We can mostly just copy / paste tests from the method that takes in the array, but passing in a single product instead. 
The intent isn't really to test the _current_ implementation, since we already know it works, but to protect us from breaking the implementation in the future without realizing